contracts/pair_anchor/src/contract.rs

```diff
@@ -0,0 +1,839 @@
+        QueryMsg::Pool {} => to_binary(&query_pool(deps)?),
+        QueryMsg::Share { amount } => to_binary(&query_share(deps, amount)?),
+        QueryMsg::Simulation { offer_asset } => to_binary(&query_simulation(deps, offer_asset)?),
+        QueryMsg::ReverseSimulation { ask_asset } => {
+            to_binary(&query_reverse_simulation(deps, ask_asset)?)
+        }
+        QueryMsg::CumulativePrices {} => to_binary(&query_cumulative_prices(deps, env)?),
+        QueryMsg::Config {} => to_binary(&query_config(deps)?),
+    }
+}
+
+/// ## Description
+/// Returns information about the pair contract in an object of type [`PairInfo`].
+/// ## Params
+/// * **deps** is an object of type [`Deps`].
+pub fn query_pair_info(deps: Deps) -> StdResult<PairInfo> {
+    let config: Config = CONFIG.load(deps.storage)?;
+    Ok(config.pair_info)
+}
+
+/// ## Description
+/// Returns the amounts of assets in the pair contract as well as the amount of LP
+/// tokens currently minted in an object of type [`PoolResponse`].
+/// ## Params
+/// * **deps** is an object of type [`Deps`].
+pub fn query_pool(deps: Deps) -> StdResult<PoolResponse> {
+    let config: Config = CONFIG.load(deps.storage)?;
+    let (assets, total_share) = pool_info(deps, config)?;
+
+    let resp = PoolResponse {
+        assets,
+        total_share,
+    };
+
+    Ok(resp)
+}
+
+/// ## Description
+/// Returns the amount of assets that could be withdrawn from the pool using a specific amount of LP tokens.
+/// The result is returned in a vector that contains objects of type [`Asset`].
+/// ## Params
+/// * **deps** is an object of type [`Deps`].
+///
+/// * **amount** is an object of type [`Uint128`]. This is the amount of LP tokens for which we calculate associated amounts of assets.
+pub fn query_share(_deps: Deps, _amount: Uint128) -> StdResult<Vec<Asset>> {
+    // let config: Config = CONFIG.load(deps.storage)?;
+    // let (pools, total_share) = pool_info(deps, config)?;
+    // let refund_assets = get_share_in_assets(&pools, amount, total_share);
+
+    Ok(vec![])
+}
+
+/// ## Description
+/// Returns information about a swap simulation in a [`SimulationResponse`] object.
+/// ## Params
+/// * **deps** is an object of type [`Deps`].
+///
+/// * **offer_asset** is an object of type [`Asset`]. This is the asset to swap as well as an amount of the said asset.
+pub fn query_simulation(deps: Deps, offer_asset: Asset) -> StdResult<SimulationResponse> {
+    let config: Config = CONFIG.load(deps.storage)?;
+    let pools: [AssetInfo; 2] = config.pair_info.asset_infos;
+
+    if offer_asset.info.equal(&pools[0]) || offer_asset.info.equal(&pools[1]) {
+    } else {
+        return Err(StdError::generic_err(
+            "Given offer asset doesn't belong to pairs",
+        ));
+    }
+
+    let result: StateResponse = deps.querier.query_wasm_smart(
+        config.anchor_market_addr,
+        &AnchorQueryMsg::State { block_height: None },
```

Comment:
You need to use [EpochState](https://github.com/Anchor-Protocol/money-market-contracts/blob/7fac132b9533f7450dfc2768ccae9318f4565d62/contracts/market/src/contract.rs#L570-L569) because it uses the same function to calculate exchange rate as [DepositStable](https://github.com/Anchor-Protocol/money-market-contracts/blob/fd70cc551dbe81d655cabf09808203fa88c0c38a/contracts/market/src/deposit.rs#L47-L46)

Reply:
done